lib/client/workloadendpoint.go

```diff
@@ -63,7 +64,16 @@ func (w *workloadEndpoints) Apply(a *api.WorkloadEndpoint) (*api.WorkloadEndpoin
 
 // Delete deletes an existing workload endpoint.
 func (w *workloadEndpoints) Delete(metadata api.WorkloadEndpointMetadata) error {
-	return w.c.delete(metadata, w)
+	// Validate the supplied Metadata.
+	if err := validator.Validate(metadata); err != nil {
+		return err
+	}
+
+	k, err := w.convertMetadataToKey(metadata)
+	if err != nil {
+		return err
+	}
+	return w.c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision})
```

Comment:
im mostly worried about other things using this. There might be things we havent' considered that might break, for example something "working" even if it was not doing a proper CaS might break with this. Not sure